x-pack/plugins/task_manager/server/task_claimers/strategy_mget.ts

```diff
@@ -0,0 +1,366 @@
+  }
+
+  // separate update for removed tasks; shouldn't happen often, so unlikely
+  // a performance concern, and keeps the rest of the logic simpler
+  let removedCount = 0;
+  if (removedTasks.size > 0) {
+    const tasksToRemove = Array.from(removedTasks);
+    for (const task of tasksToRemove) {
+      task.status = TaskStatus.Unrecognized;
+    }
+
+    // don't worry too much about errors, we'll get them next time
+    try {
+      const removeResults = await taskStore.bulkUpdate(tasksToRemove, { validate: false });
+      for (const removeResult of removeResults) {
+        if (isOk(removeResult)) {
+          removedCount++;
+        } else {
+          const { id, type, error } = removeResult.error;
+          logger.warn(
+            `Error updating task ${id}:${type} to mark as unrecognized during claim: ${error.message}`,
+            logMeta
+          );
+        }
+      }
+    } catch (err) {
+      logger.warn(`Error updating tasks to mark as unrecognized during claim: ${err}`, logMeta);
+    }
+  }
+
+  const message = `task claimer claimed: ${finalResults.length}; stale: ${staleTasks.size}; conflicts: ${conflicts}; missing: ${missingTasks.size}; updateErrors: ${bulkErrors}; removed: ${removedCount};`;
+  logger.debug(message, logMeta);
+
+  const finalResult = {
+    stats: {
+      tasksUpdated: finalResults.length,
+      tasksConflicted: conflicts,
+      tasksClaimed: finalResults.length,
+    },
+    docs: finalResults,
+    timing: stopTaskTimer(),
+  };
+
+  for (const doc of finalResults) {
+    events$.next(asTaskClaimEvent(doc.id, asOk(doc), finalResult.timing));
+  }
+
+  return finalResult;
+}
+
+interface SearchAvailableTasksResponse {
+  docs: ConcreteTaskInstance[];
+  versionMap: Map<string, ConcreteTaskInstanceVersion>;
+}
+
+async function searchAvailableTasks({
+  definitions,
+  taskTypes,
+  removedTypes,
+  excludedTypes,
+  taskStore,
+  claimOwnershipUntil,
+  size,
+  taskMaxAttempts,
+}: OwnershipClaimingOpts): Promise<SearchAvailableTasksResponse> {
+  const searchedTypes = Array.from(taskTypes)
+    .concat(Array.from(removedTypes))
+    .filter((type) => !excludedTypes.has(type));
+  const queryForScheduledTasks = mustBeAllOf(
+    // Task must be enabled
+    EnabledTask,
+    // a task type that's not excluded (may be removed or not)
+    OneOfTaskTypes('task.taskType', searchedTypes),
+    // Either a task with idle status and runAt <= now or
+    // status running or claiming with a retryAt <= now.
+    shouldBeOneOf(IdleTaskWithExpiredRunAt, RunningOrClaimingTaskWithExpiredRetryAt),
+    // must have a status that isn't 'unrecognized'
+    RecognizedTask
+  );
+
+  const sort: NonNullable<SearchOpts['sort']> = getClaimSort(definitions);
+  const query = matchesClauses(queryForScheduledTasks, filterDownBy(InactiveTasks));
+
+  return await taskStore.fetch({
+    query,
+    sort,
+    size,
+    seq_no_primary_term: true,
+  });
```

Comment:
Opening this comment so we don't forget to create a follow up issue to run separate queries for the task types that have concurrencies vs the rest.

Reply:
ya, it's in the top comment TODO's list

Reply:
Ah, I completely missed it 🙈 resolving..